python/mpt/src/mpt/utils.cairo

```diff
@@ -189,8 +189,72 @@ func check_branch_node(node: BranchNode) {
             raise('ValueError');
         }
     }
+
     return ();
 }
+
+func check_leaf_node(path: Bytes, node: LeafNode) {
+    alloc_locals;
+
+    let nibbles_len = node.value.rest_of_key.value.len;
+    let path_len = path.value.len;
+
+    if (nibbles_len + path_len != 64) {
+        raise('ValueError');
+    }
+
+    let bytes_variant = node.value.value.value.bytes.value;
+    if (cast(bytes_variant, felt) != 0) {
+        if (bytes_variant.len == 0) {
+            raise('ValueError');
+        }
+        return ();
+    }
+
+    let sequence_variant = node.value.value.value.sequence.value;
+    if (cast(sequence_variant, felt) != 0) {
+        if (sequence_variant.len == 0) {
+            raise('ValueError');
+        }
+        return ();
+    }
+
+    with_attr error_message("ValueError: Unsupported leaf value variant") {
+        jmp raise.raise_label;
+    }
+}
+
+func check_extension_node(node: ExtensionNode) {
```

Comment:
nice:)

Reply:
i'm realizing as we speak that also you're not allowed to have an extension node point to another extension node, not sure how we can implement that

Reply:
this would need to be a check made when running `compute_diff`: you'd need to pass the parent nodes, and ensure the resolved child is not an extension

Reply:
https://github.com/kkrt-labs/keth/issues/1269